File: src/hao_common/src/circle_pub.cpp
/*
    编写 ROS 节点，控制小乌龟画圆

    准备工作:
        1.获取topic(已知: /turtle1/cmd_vel)
        2.获取消息类型(已知: geometry_msgs/Twist)
        3.运行前，注意先启动 turtlesim_node 节点

    实现流程:
        1.包含头文件
        2.初始化 ROS 节点
        3.创建发布者对象
        4.循环发布运动控制消息
*/
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
int main(int argc, char * argv[]){
    setlocale(LC_ALL, "");
    ros::init(argc, argv, "control");
    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel",1000);
    geometry_msgs::Twist msg;
    msg.linear.x = 1;
    msg.angular.z = 1;

    ros::Rate r(10);
    while(ros::ok()){
        pub.publish(msg);
        ros::spinOnce();
    }
    return 0;
}

File: src/hao_common/scripts/circle_pub.py
#! /usr/bin/env python
import rospy
from geometry_msgs.msg import Twist

if __name__ == "__main__":
    rospy.init_node("circle_pub")

    pub = rospy.Publisher("/turtle1/cmd_vel",Twist,queue_size=1000)
    rate = rospy.Rate(10)
    msg = Twist()
    msg.linear.x = 2.0
    msg.angular.z = 2.0

    while not rospy.is_shutdown():
        pub.publish(msg)
        rate.sleep()